sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/util/DbUtils.java

```diff
@@ -41,44 +48,108 @@
+        .createAlias("location3.parentLocation", "location4", JoinType.LEFT_OUTER_JOIN)
+        .createAlias("location4.parentLocation", "location5", JoinType.LEFT_OUTER_JOIN)
+        .createAlias("location5.parentLocation", "location6", JoinType.LEFT_OUTER_JOIN)
+        .add(Restrictions.or(
+            Restrictions.and(Restrictions.eq("location1.locationUnit", LocationUnit.FREEZER),
+                textRestriction("location1.alias", sanitized, matchMode)),
+            Restrictions.and(Restrictions.eq("location2.locationUnit", LocationUnit.FREEZER),
+                textRestriction("location2.alias", sanitized, matchMode)),
+            Restrictions.and(Restrictions.eq("location3.locationUnit", LocationUnit.FREEZER),
+                textRestriction("location3.alias", sanitized, matchMode)),
+            Restrictions.and(Restrictions.eq("location4.locationUnit", LocationUnit.FREEZER),
+                textRestriction("location4.alias", sanitized, matchMode)),
+            Restrictions.and(Restrictions.eq("location5.locationUnit", LocationUnit.FREEZER),
+                textRestriction("location5.alias", sanitized, matchMode)),
+            Restrictions.and(Restrictions.eq("location6.locationUnit", LocationUnit.FREEZER),
+                textRestriction("location6.alias", sanitized, matchMode))));
   }
 
-  /**
-   * Create a Hibernate criterion to search for all the properties our users want to search.
-   *
-   * @param querystr
-   * @return
-   */
-  public static Criterion searchRestrictions(String querystr, boolean exact, String... searchProperties) {
-    String str = DbUtils.convertStringToSearchQuery(querystr, exact);
+  public static void restrictPaginationByDistributionRecipient(Criteria criteria, TextQuery query, String collectionProperty,
+      String itemIdProperty) {
+    if (query.getText() == null) {
+      DetachedCriteria subquery = DetachedCriteria.forClass(ListTransferView.class)
+          .createAlias(collectionProperty, "transferItem")
+          .add(Restrictions.isNotNull("recipient"))
+          .setProjection(Projections.property("transferItem." + itemIdProperty));
+      criteria.add(Property.forName("id").notIn(subquery));
+    } else {
+      criteria.createAlias("listTransferViews", "transfer")
+          .add(DbUtils.textRestriction(query, "transfer.recipient"));
+    }
+  }
+
+  public static Criterion textRestriction(TextQuery query, String... searchProperties) {
+    String sanitized = sanitizeQueryString(query.getText());
+    MatchMode matchMode = getMatchMode(query);
+    if (searchProperties.length == 1) {
+      return textRestriction(searchProperties[0], sanitized, matchMode);
+    } else {
+      Criterion[] criteria = new Criterion[searchProperties.length];
+      for (int i = 0; i < searchProperties.length; i++) {
+        criteria[i] = textRestriction(searchProperties[i], sanitized, matchMode);
+      }
+      if (sanitized == null) {
```

Comment:
This is slightly confusing...if the query is `IS NULL` and we have multiple properties, then it becomes _all are null_, but if we're searching for text, then _any field has this text_?

Reply:
For searches that check multiple fields, if you're looking for nulls, it should be null in all those fields. e.g. "external" checks both `externalName` and `secondaryIdentifier`. Searching "external:" finds items with neither specified this way. If it were "any is null" then the query would always return everything.

Reply:
Makes sense. Can you stick this in a comment?

Reply:
done